### DISCUSSION TITLE
Api Rules based on a roles field

### DISCUSSION THREAD

**Discussion by FrenchFryInteractive** [pocketbase/pocketbase#3514] [Q&A]
Hey, how can I use API rules in a collection called posts and ensure that only the creator and someone who has admin in the roles field. Anyone who can tell me how to do that?

**Comment by ganigeorgiev** [comment]
Assuming that "roles" is a multiple `select` field and "creator" is a single `relation` to the "users" collection, you can try with:

```js
@request.auth.id != "" &&
(@request.auth.id = creator || @request.auth.roles:each ?= "admin")
```
The above roughly translates to:
_Require the user to be authenticated AND their id must either be equal to the "creator" id OR at-least-one of their "roles" field values must match to "admin"._
